### ISSUE TITLE
No epoch when training ResNet50 Classification

### ISSUE THREAD

**Issue by AnhPC03** [AlexeyAB/darknet#6663]
Hello @AlexeyAB, I've train cat and dog classification with ResNet50 backbone.
I used resnet50.cfg, and resnet50.weights downloaded from pjreddie website.
And run command ``` ./darknet -i 1 classifier train cfg/custom.data cfg/resnet50.cfg resnet50.weights```
I run on my GPU.
My **custom.data** file:
```
classes=2
train=../cat_dog_own/train.list
valid=../cat_dog_own/test.list
labels = ../cat_dog_own/labels.txt
backup=backup/
```
And I changed last convolution layer in **resnet50.cfg** file to ```filters=2```
And I got result as image:
![Screenshot from 2020-09-15 10-25-21](https://user-images.githubusercontent.com/35995951/93164092-e4fe6180-f742-11ea-98b6-b3d80a484a34.png)
As you can see, has no training epoch. If I use resnet.weights saved in backup as in the image, result seems random 50:50.
How I can train cat and dog classification with ResNet50 backbone?

**Comment by zhoukaii** [comment]
![image](https://user-images.githubusercontent.com/22705512/93174367-6fa18980-f760-11ea-8a06-2ab80b7725d0.png)
how could you do that 
i even cannot train when i use resnet50

**Comment by AnhPC03** [comment]
@zhoukaii 
Comment the 981st line in ***./src/parser.c** file
The 981st line is: 
```
assert(params.w == net.layers[index].out_w && params.h == net.layers[index].out_h);
```

**Comment by wanfade** [comment]
Try to change max_batches=80000 in the resnet.cfg bigger.
I set max_batches to 8000000000000, the training begin normally.

**Comment by AnhPC03** [comment]
@wanfade I've changed as you said and then trained model.
But the loss was inf as below image
![Screenshot from 2020-09-16 13-53-14](https://user-images.githubusercontent.com/35995951/93302513-5235f380-f824-11ea-843b-54f6628643e1.png)

**Comment by wanfade** [comment]
Some experience for small dataset.
It's recommend to extract weight from resnet50.weights.
`./darknet partial cfg/resnet50.cfg resnet50.weights resnet50.conv.65 65`
Then begin training from scratch with the initial weight file 'resnet50.conv.65'.
If training loss still too big , just change loss to 0.001 or smaller.
Delete the 'burns' in the cfg file will also help you.

**Comment by AnhPC03** [comment]
@wanfade how can i *change loss to 0.001*?

**Comment by wanfade** [comment]
In the resnet50.cfg : 
change below 
```
burn_in=1000
learning_rate=0.1
```
to
```
#burn_in=1000
learning_rate=0.001
```

**Comment by AnhPC03** [comment]
@wanfade I've changed as you said and it effected in some beginning epochs. Then it've gone inf. Do you had train the resnet50 classification network converged?
![Screenshot from 2020-09-16 14-34-25](https://user-images.githubusercontent.com/35995951/93306568-43523f80-f82a-11ea-97c0-34de9c6871a3.png)

**Comment by wanfade** [comment]
I trained resnet50 from scratch with weights which is extracted from resnet50.weights.
And it converged in 14 epochs on a 2-classes dataset with avg loss 0.05.

**Comment by AnhPC03** [comment]
Thanks, @wanfade . Could you explain what does ```./darknet partial``` mean please?

**Comment by wanfade** [comment]
Example:  darknet partial cfg/yolov4-tiny-custom.cfg yolov4-tiny.weights yolov4-tiny.conv.29 29
This command will extract 29 convolution layers weights from the given weight file ' yolov4-tiny.weights '. 
And then you can train with yolov4-tiny.conv.29 which means you train a network from a initial weight in the first 29 convolution layers and random weights in the rest layers.

**Comment by AnhPC03** [comment]
@wanfade Thanks for your explain. I've done in that way, it trained normally, but the loss was always in range 0.9 to 1.1. it didn't decrease much, although i've set learning rate to 0.001, even 0.0001.
Do you have any solution?

**Comment by wanfade** [comment]
The speed of convergence depends on many factors. 
If loss decrease slowly, maybe a higher learning rate is needed. 
But keep the training for a period and  i think the loss will go down.

**Comment by AnhPC03** [comment]
The loss oscillate in range (0.9, 1.1). But anyway, many thanks, @wanfade . I'll try to change some parameters to converge model.
